api/v1alpha1/selfnoderemediationconfig_webhook.go

```diff
@@ -172,3 +175,26 @@ func validateToleration(toleration v1.Toleration) error {
 	}
 	return nil
 }
+
+func (r *SelfNodeRemediationConfig) validateMinRebootTime() error {
+	if r.Status.MinSafeTimeToAssumeNodeRebootedSeconds == 0 {
+		return fmt.Errorf("failed to verify min value of SafeRebootTimeSec, Status.MinSafeTimeToAssumeNodeRebootedSeconds should not be empty")
+	}
+
+	//allow removing this optional field
+	if r.Spec.SafeTimeToAssumeNodeRebootedSeconds == 0 {
+		return nil
+	}
+
+	if r.Status.MinSafeTimeToAssumeNodeRebootedSeconds > r.Spec.SafeTimeToAssumeNodeRebootedSeconds {
+		return fmt.Errorf("can not set SafeTimeToAssumeNodeRebootedSeconds value below the calculated minimum value of: %d", r.Status.MinSafeTimeToAssumeNodeRebootedSeconds)
+	}
+	return nil
+}
+
+func (r *SelfNodeRemediationConfig) validateSafeTimeEmpty() error {
+	if r.Spec.SafeTimeToAssumeNodeRebootedSeconds > 0 {
+		return fmt.Errorf("SafeTimeToAssumeNodeRebootedSeconds can only be set after configuration is created")
```

Comment:
My line of thinking was: at this stage we can prevent the user from creating an invalid configuration which is a better choice than dealing with it later.
In context of your [comment](https://github.com/medik8s/self-node-remediation/pull/197#discussion_r1601345595) regarding multiple configs, I think that assuming we will only allow the initial default config to be created this would indeed be redundant.

ATM since in any case if we have multiple configurations an invalid `Spec.SafeTimeToAssumeNodeRebootedSeconds` is probably the lease of our worries, I don't mind removing it to simplify the code.